cli/src/pcluster/templates/cdk_builder_utils.py

```diff
@@ -250,17 +251,59 @@ def get_queue_security_groups_full(managed_compute_security_group: ec2.CfnSecuri
     return queue_security_groups
 
 
+def add_cluster_iam_resource_prefix(stack_name, config, name: str, iam_type: str):
+    """Return a path and Name prefix from the Resource prefix config option."""
+    full_resource_path = None
+    full_resource_name = None
+    if config.iam and config.iam.resource_prefix:
+        iam_path, iam_name_prefix = get_path_n_name_prefix_from_iam_resource_prefix(config.iam.resource_prefix)
+        if iam_name_prefix:
+            if iam_type in ["AWS::IAM::Role", "AWS::IAM::InstanceProfile"]:
+                full_resource_name = iam_name_prefix + name + "-"
```

Comment:
You should break this up to get all the individual pieces first and assign it once to `full_resource_name` e.g. 

```
part_a = x
part_b = y
part_c = z + a + f
full_name = part_a + part_b + part_c
```

I think this will be easier to digest

Reply:
Made it more readable friendly